clients/src/main/java/org/apache/kafka/clients/ClusterConnectionStates.java

```diff
@@ -440,6 +441,20 @@ public boolean isConnectionSetupTimeout(String id, long now) {
         return now - lastConnectAttemptMs(id) > connectionSetupTimeoutMs(id);
     }
 
+    /**
+     * Return the Set of nodes whose connection setup has timed out.
+     * @param now the current time in ms
+     */
+    public Set<String> nodesWithConnectionSetupTimeout(long now) {
+        Set<String> nodes = new HashSet<>();
+        for (String nodeId : connectingNodes) {
```

Comment:
We can use `connectingNodes.stream().filter`?

Reply:
Sure thing.